src/runtime/virtcontainers/kata_agent.go

```diff
@@ -722,30 +729,38 @@ func (k *kataAgent) startSandbox(ctx context.Context, sandbox *Sandbox) error {
 		return err
 	}
 
-	// Check grpc server is serving
-	if err = k.check(ctx); err != nil {
-		return err
-	}
+	var kmodules []*grpc.KernelModule
 
-	// Setup network interfaces and routes
-	interfaces, routes, neighs, err := generateVCNetworkStructures(ctx, sandbox.network)
-	if err != nil {
-		return err
-	}
-	if err = k.updateInterfaces(ctx, interfaces); err != nil {
-		return err
-	}
-	if _, err = k.updateRoutes(ctx, routes); err != nil {
-		return err
-	}
-	if err = k.addARPNeighbors(ctx, neighs); err != nil {
-		return err
+	if sandbox.config.HypervisorType == RemoteHypervisor {
+		ctx = context.WithValue(ctx, customRequestTimeoutKey, remoteRequestTimeout)
+	} else {
+		// TODO: Enable the following features for remote hypervisor if necessary
+
+		// Check grpc server is serving
+		if err = k.check(ctx); err != nil {
+			return err
+		}
+
+		// Setup network interfaces and routes
+		interfaces, routes, neighs, err := generateVCNetworkStructures(ctx, sandbox.network)
+		if err != nil {
+			return err
+		}
+		if err = k.updateInterfaces(ctx, interfaces); err != nil {
+			return err
+		}
+		if _, err = k.updateRoutes(ctx, routes); err != nil {
+			return err
+		}
+		if err = k.addARPNeighbors(ctx, neighs); err != nil {
+			return err
+		}
+
+		kmodules = setupKernelModules(k.kmodules)
 	}
 
 	storages := setupStorages(ctx, sandbox)
```

Comment:
Sorry to ask: does the storage setup keeps the same for remote hypervisor?

Reply:
[`setupStorage`](https://github.com/kata-containers/kata-containers/blob/6656c06924cc3b137f2086342538edaaeb851a96/src/runtime/virtcontainers/kata_agent.go#L804-L872) does nothing for remote hypervisor, since it does not support file system sharing.

I will change the code to call `setupStorage` unless remote hypervisor is used.

Reply:
I investigated the code, and found that `setupStorage` must be called even for remote hypervisor, so I keep the old code.

The following code needs to be called to prepare shm device on pod vm.
https://github.com/kata-containers/kata-containers/blob/f30931cda97fbbf6253eac97ed2858327ae9493e/src/runtime/virtcontainers/kata_agent.go#L851-L862